lib/FrontendTool/FrontendTool.cpp

```diff
@@ -685,6 +685,19 @@ static SourceFile *getPrimaryOrMainSourceFile(CompilerInvocation &Invocation,
   return SF;
 }
 
+static void dumpAST(CompilerInvocation &Invocation, CompilerInstance &Instance) {
+  SourceFile *sourceFile = getPrimaryOrMainSourceFile(Invocation, Instance);
```

Comment:
Hm. This isn't going to work in batch mode, where you have more than one primary source file. Take a look at `emitLoadedModuleTraceForAllPrimariesIfNeeded` instead.